lib/msal-angular/src/msal.interceptor.ts

```diff
@@ -116,36 +117,59 @@ export class MsalInterceptor implements HttpInterceptor {
-     * @param protectedResourcesArray 
+     * @param protectedResourcesEndpoints
      * @param endpoint 
-     * @param location 
      * @returns 
      */
-    private matchResourcesToEndpoint(protectedResourcesEndpoints: string[], endpoint: string): Array<string> {
-        return protectedResourcesEndpoints.filter(key => {
+    private matchResourcesToEndpoint(protectedResourcesEndpoints: string[], endpoint: string): MatchingResources {
+        const matchingResources: MatchingResources = {absoluteResources: [], relativeResources: []};
+
+        protectedResourcesEndpoints.forEach(key => {
+            // Normalizes and adds resource to matchingResources.absoluteResources if key matches endpoint. StringUtils.matchPattern accounts for wildcards
             const normalizedKey = this.location.normalize(key);
+            if (StringUtils.matchPattern(normalizedKey, endpoint)){
+                matchingResources.absoluteResources.push(key);
+            }
             
+            // Get url components for relative urls
+            const absoluteKey = this.getAbsoluteUrl(key);
+            const keyComponents = new UrlString(absoluteKey).getUrlComponents();
+            const absoluteEndpoint = this.getAbsoluteUrl(endpoint);
+            const endpointComponents = new UrlString(absoluteEndpoint).getUrlComponents();
+
             // Normalized key should include query strings if applicable
-            const keyComponents = new UrlString(key).getUrlComponents();
             const relativeNormalizedKey = keyComponents.QueryString ? `${keyComponents.AbsolutePath}?${keyComponents.QueryString}` : this.location.normalize(keyComponents.AbsolutePath);
 
-            // Relative endpoint not applicable, matching endpoint with protected resource. StringUtils.matchPattern accounts for wildcards
-            if (relativeNormalizedKey === "" || relativeNormalizedKey === "/*") {
-                return StringUtils.matchPattern(normalizedKey, endpoint);
-            } else {
-                // Matching endpoint with both protected resource and relative url of protected resource
-                return StringUtils.matchPattern(normalizedKey, endpoint) || StringUtils.matchPattern(relativeNormalizedKey, endpoint);
+            // Add resource to matchingResources.relativeResources if same origin, relativeKey matches endpoint, and is not empty
+            if (keyComponents.HostNameAndPort === endpointComponents.HostNameAndPort && StringUtils.matchPattern(relativeNormalizedKey, absoluteEndpoint) && relativeNormalizedKey !== "" && relativeNormalizedKey !== "/*"){
+                matchingResources.relativeResources.push(key);
             }
         });
+
+        return matchingResources;
+    }
+
+    /**
+     * Transforms relative urls to absolute urls
+     * @param url 
+     * @returns 
+     */
+    private getAbsoluteUrl(url: string): string {
+        const link = document.createElement("a");
```

Comment:
After following the instructions above: 
```suggestion
        const link = this.document.createElement("a");
```